core/src/util/ext-tools.ts

```
@@ -202,9 +201,8 @@ export class CliWrapper {
   }
 }
 
-interface PluginToolOpts {
-  platform?: string
-  architecture?: string
+const findBuildSpec = (spec: PluginToolSpec, plat: string, arch: string) => {
+  return spec.builds.find((build) => build.platform === plat && build.architecture === arch)!
```

I don't think using `!` here is a good idea, because due to our error we might have a case where nothing is found. Instead maybe just throw a helpful error if the result is undefined.

---

it's actually being checked in line 244. that check works, I tested it - there is a nice user friendly error message.

---

I can eliminate the `!` here, but then I can't assign directly to `this.buildspec` in line  239 and I'd need to introduce another variable. Would you prefer that?

---

I moved it to lines 239 and 241, then it might be more clear. do you agree?